mp/src/game/shared/momentum/run/mom_replay_factory.h

```diff
@@ -17,8 +24,50 @@ class CMomReplayFactory
 
     // Returns a replay file and constructs a versioned replay object.
     CMomReplayBase *LoadReplayFile(const char *pFileName, bool bFullLoad = true, const char *pPathID = "MOD");
+    CMomReplayBase *LoadReplayFromBuffer(CUtlBuffer& reader, bool bFullLoad = true);
 
     uint8 m_ucCurrentVersion;
+
+    /* Asynchronous replay loading
+     * The interaction should look like this:
+     * call AsyncLoadReplayFileMultiple -> have the callback send a message to recipient
+     * -> get replay files by calling AsyncGetReplays
+     */
+    // Client-only for now
+#ifdef CLIENT_DLL
+
+
+    LocalTimesStatus_t AsyncLoadReplayFileMultiple(const CUtlStringList& filenames, const char *pathID,
+    MomLocalReplayCallback_t callback, MomReplayAsyncHandle* outHandle = nullptr);
+    LocalTimesStatus_t AsyncCancel(const MomReplayAsyncHandle& handle);
+
+    // Returns all loaded files if finished.
+    // If not, do nothing or give partial result based on bCancelRemaining
+    LocalTimesStatus_t AsyncGetReplays(MomReplayAsyncHandle handle, CUtlVector<CMomReplayBase*>& vecToReturn, bool bCancelRemaining = false);
+
+    void AsyncCleanup(const MomReplayAsyncHandle& handle);
+
+private:
+    struct MomReplayLoadRequestInternal
+    {
+        MomReplayAsyncHandle                handle;
+        MomLocalReplayCallback_t callback;
+        FSAsyncControl_t*                   arrControls;    // We keep these so we can abort the requests if needed
```

Comment:
Why not also a CUtlVector? Or a CUtlLinkedList?

Reply:
When looking at filesystem.h for too long you really get caught up in raw C, I'll switch this to containers.